src/mono/mono/component/marshal-ilgen-stub.c

```diff
@@ -14,6 +14,37 @@ stub_emit_marshal_ilgen (EmitMarshalContext *m, int argnum, MonoType *t,
 		MonoMarshalSpec *spec, int conv_arg,	
 		MonoType **conv_arg_type, MarshalAction action, MonoMarshalLightweightCallbacks* lightweight_cb)
 {
+	if (spec && spec->native == MONO_NATIVE_CUSTOM)
+		return conv_arg;
+
+	if (spec && spec->native == MONO_NATIVE_ASANY)
+		return conv_arg;
```

Comment:
Lightweight marshaler shouldn't see any custom marshalers or AsAny.  I wonder if we can assert for these.